### PR TITLE
19 add drawing language

### PR DESCRIPTION
**Pull request by starsasumi**
This is for issue #19

With this language, we can draw things with much simple codes.

### CODE REVIEW

File: src/view/Painting.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import language.DrawingTask;
+import language.Painter;
+
+import java.awt.Color;
+
+/**
+ * Created by Xiaxing SHI on 21/11/15.
+ */
+public class Painting implements DrawingTask {
```
Comment (starsasumi) [inline src/view/Painting.java:14]: This class is the language we defined. User who prefers writing codes could use this class to draw rather than use directly our Shape objects

**Comment by anqitong** [comment]
Don't get it. What's your point?

**Comment by anqitong** [comment]
@starsasumi : problem in the test for #67 :

```
in the window the polyline is not filled in black
in the svgAndAwt.svg file the polyline is filled in black!
```

**Comment by starsasumi** [comment]
@anqitong As the prof required, `Painting.java` is the "high-level language" that he wants. And codes in our presentations would be the "middle-level language"

**Comment by anqitong** [comment]
So you're suggesting taht we use directly java in Painting as high-level language?

**Comment by starsasumi** [comment]
Yes, and the prototype of this `Painting` is in our discussion in #19

**Comment by anqitong** [comment]
I think using shape objetcts is already a high-level language.

**Comment by starsasumi** [comment]
With this, we now have a java-based high-level language, and also a GUI for drawing. ;)

**Comment by anqitong** [comment]
I'm not sure it is necessary, but if you really want to use Painting as high-level language, I'm not against it. So I would like to know what @Thomas-dot-G  thinks

**Comment by starsasumi** [comment]
@anqitong I understand, I am also not sure about this. This is why I re-open the issue to ask you two. But I think maybe make it as real RP would be more clear to discuss. 

As you can see, codes in `Painting` are much more clean. And with this, we have a simple-to-use high-level language and also a great GUI, which make our project rocks ;)

**Comment by anqitong** [comment]
I think you're right about opening a PR to make it clearer!

File: src/language/BasicPainter.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    @Override
+    public void circle(int x, int y, int r) {
+        Circle c = new Circle(30,30,30);
+        addToShapes(c);
+    }
+
+    @Override
+    public void ellipse(int x, int y, int rx, int ry) {
+        Ellipse e = new Ellipse(100, 30, 40, 20);
```
Comment (Thomas-dot-G) [inline src/language/BasicPainter.java]: Each time, you should put the arguments rather than random int.

File: src/test/TestLanguage.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import model.presentation.xml.SVGFile;
+import language.BasicPainter;
+
+/**
+ * This could be the final main class in the project. It read user's drawing task and draw it. Then it shows the
+ * painting in JFrame and in a SVG file.
+ *
+ * Created by Xiaxing SHI on 21/11/15.
+ */
+public class TestLanguage {
```
Comment (Thomas-dot-G) [inline src/test/TestLanguage.java:17]: Could you make it a TestClass for now? we will change the actual main eventually after.
(What I mean is, add stuff, do some asserts)

File: src/language/BasicPainter.java (hunk 2 above)
Comment (starsasumi) [inline src/language/BasicPainter.java]: Ah! I am silly ...

File: src/view/Painting.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Edit this method to draw whatever you want!
+     * But you should only the parameter p.
```
Comment (Thomas-dot-G) [inline src/view/Painting.java:17]: `should only the parameter` need some correction

File: src/test/TestLanguage.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        BasicPainter painter = new BasicPainter();
+        Painting painting = new Painting();
+        painting.draw(painter);
+
+        // Create an awt presentation to draw
+        AWTFrame awt = new AWTFrame(painter.getName(), painter.getCanvas());
+        awt.addShapes(painter.getShapes());
+        awt.createDrawing();
+
+        // all presentation should also work, so we test them
```
Comment (Thomas-dot-G) [inline src/test/TestLanguage.java]: (Just `all presentations`)

File: src/view/Painting.java (hunk 1 above)
Comment (Thomas-dot-G) [inline src/view/Painting.java:14]: I thing the user should write its code in only one place, even if he wants to use both methods to write what he wants to draw

File: src/test/TestLanguage.java (hunk 3 above)
Comment (starsasumi) [inline src/test/TestLanguage.java:17]: @Thomas-dot-G Ok, I will do it later tonight. Like adding some assert to compare xml results

**Comment by Thomas-dot-G** [comment]
I don't think this is required, but it can be nice to have a simpler code for the user.
However, that should be well documented for M Grall.

**Comment by starsasumi** [comment]
@Thomas-dot-G If you two agree to merge this, I will rewrite the README.md first ;)

**Comment by Thomas-dot-G** [comment]
Not yet to merge, but I agree to merge it later (after some modifications)

**Comment by starsasumi** [comment]
Hey guys, I add tests in `TestLanguage.java`. Is it better now?

File: src/language/Painter.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package language;
+
+import model.specific_path.LineType;
+import java.awt.*;
+
+/**
+ * This defined what a user can do for drawing things
```
Comment (Thomas-dot-G) [inline src/language/Painter.java]: this defines (minor)

File: src/view/Draw.java [hunk 7]
```diff
@@ -1,171 +1,26 @@
...
-		// Set the pencil
-		r.setPencil(p);
-		
-		// Fill it
-		r.setColor(new ColorSVG(Color.green.getRGB()));
+		BasicPainter painter = new BasicPainter();
+		Painting painting = new Painting();
+		painting.draw(painter);
 
-		// Gather all the shapes
```
Comment (Thomas-dot-G) [inline src/view/Draw.java:44]: Maybe keep the two way of doing it, it might be more useful if we have a problem with either one of them.

And you also removed the comment without adding new ones. Could you do that ?

File: src/view/Painting.java (hunk 4 above)
Comment (Thomas-dot-G) [inline src/view/Painting.java:17]: still on

File: src/view/Painting.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        p.stroke(1, Color.blue);
+        p.fill(Color.red);
+        p.circle(30, 30, 30);
+
+        p.stroke(2, Color.darkGray);
+        p.ellipse(100, 30, 40, 20);
+
+        p.line(5, 60, 15, 100);
+
+        p.polygone(new int[]{100,200,100}, new int[]{100, 200, 200});
```
Comment (Thomas-dot-G) [inline src/view/Painting.java]: for polygone and polyline you could have used a polygone(Points... points), thus you can add as many arguments as you want. It might be simpler for the user

Comment (starsasumi) [inline src/view/Painting.java]: @Thomas-dot-G I thinked about this. And for myself, I think it's easier to write two inline int arrays than to create an array of Point and then new some points.

@anqitong What do you think?

File: src/view/Draw.java (hunk 7 above)
Comment (starsasumi) [inline src/view/Draw.java:44]: Ok, I will later create another main class for Painting ;), and add comments

**Comment by starsasumi** [comment]
I also update README.md. What do you think about it. I will add the insert back later in the `Painter` API.

File: src/view/Painting.java (hunk 8 above)
Comment (Thomas-dot-G) [inline src/view/Painting.java]: Not an array, just some points. new Polygone(point1, point2, point3)
with public Polygone(Point... points){ }

Comment (starsasumi) [inline src/view/Painting.java]: A shortcut for a triangle? I thought polygone can take as many as points.

Comment (Thomas-dot-G) [inline src/view/Painting.java]: It was an example, with Point... you can add as many points as you want. That thing in java means as many arguments as you want at the end. I used that in my PR I think.

**Comment by anqitong** [comment]
Okay for me

**Comment by anqitong** [comment]
I will have to add a paragraph descrbing how to use the GUI

File: src/view/Painting.java (hunk 8 above)
Comment (starsasumi) [inline src/view/Painting.java]: Got it, you mean the `...` argument

Comment (Thomas-dot-G) [inline src/view/Painting.java]: yes

**Comment by starsasumi** [comment]
Is this good to go now?

File: README.md [hunk 9]
```diff
@@ -1,30 +1,80 @@
...
-### Step 2:
-You'll then need some shapes to draw.
-The list of shapes are for now listed in src/model/ShapeName.
-Each type of shape has it's own attribute. So for now, you'll need to see what they are for each specific shapes.
-They are located in src/model/specific_path.
-As an example: `Circle c = new Circle(50,40,30);`
-You are encourage to add a color to your shape like that: `c.setColor(new ColorSVG(Color.blue.getRGB()));`
-As well as defining the pencil used to draw that shape, pencil you would already have created: `c.setPencil(p);`
+- `view.GUI`: This is the entrance for our GUI.
+- `view.Painting`: Here is the "drawing language" that we defined. It is a script that you could modify to tell the software what it should draw.
```
Comment (Thomas-dot-G) [inline README.md]: All this lines will form only one line if you don't skip another line.

`It is a script that you could modify to tell` could you change it into `It is a script that you can modify to tell`

Or you can do the read me in another PR actually, so this PR can be merged sooner

File: README.md [hunk 10]
```diff
@@ -1,30 +1,80 @@
...
+
+Anqi Tong
+
+Thomas Garesse
+
+Xiaxing Shi
+
+- - -
+## Sould be reused after defined insert in Painter
 ### Step 3:
```
Comment (Thomas-dot-G) [inline README.md:28]: You remove the fist steps, but kept these, so it now starts at step 3

**Comment by Thomas-dot-G** [comment]
Except for the Read Me, I am ok to merge. If you remove your changes for the read me and put them in another PR, I think we are ready to merge this.

**Comment by starsasumi** [comment]
OK
